File: ttlogix-web/App/ClientApp/src/store/models/decantItem.ts
import { DecantItemAddMod } from "./decantItemAddMod copy";
import { DecantItemPallet } from "./decantItemPallet";

export class DecantItems{
    items: Array<DecantItem>;
    jobNo: string;
}

export class DecantItem {
    constructor(preItem: DecantItemAddMod | null = null){
        if(preItem){
            this.id = 0;
            this.sourcePalletId = '';
            this.sourceQty = 0;
            this.newPallets = new Array<DecantItemPallet>();
            return this;
        }
    }
    id: number;
    sourcePalletId: string;
    sourceQty: number;
    newPallets: Array<DecantItemPallet>;
}

File: tt-api/src/Domain/Entities/DecantItem.cs
namespace Domain.Entities;

public class DecantItem
{
    public string SourcePalletId { get; set; } = null!;
    public int SourceQty { get; set; }

    public ICollection<DecantItemPallet> NewPallets { get; set; } = new List<DecantItemPallet>();
}